watcher/src/watcher.rs

```diff
@@ -191,7 +191,7 @@ impl Watcher {
                         log::info!(
                             self.logger,
                             "Archive block retrieved for {:?} {:?}",
```

Comment:
You should be able to change this to `"Archive block retrieved for {}: {}",`, I don't think the debug formatters are necessary

Reply:
Yep, this will also remove the need for `as_str()`, since `Url` implements `Display` :-)

Reply:
Sounds good, I'll do that